block/src/Pos/Arbitrary/Block/Message.hs

```diff
@@ -12,25 +12,26 @@ import           Pos.Arbitrary.Txp ()
 import           Pos.Arbitrary.Update ()
 import           Pos.Binary.Class (Bi, Raw)
 import qualified Pos.Block.Network.Types as T
-import           Pos.Core (HasConfiguration)
+import           Pos.Core (HasProtocolConstants, HasProtocolMagic, HasGenesisHash)
 import           Pos.Core.Ssc (SscPayload, SscProof)
 
 ------------------------------------------------------------------------------------------
 -- Block network types
 ------------------------------------------------------------------------------------------
 
-instance HasConfiguration => Arbitrary T.MsgGetHeaders where
+instance Arbitrary T.MsgGetHeaders where
     arbitrary = genericArbitrary
     shrink = genericShrink
 
-instance HasConfiguration => Arbitrary T.MsgGetBlocks where
+instance Arbitrary T.MsgGetBlocks where
     arbitrary = genericArbitrary
     shrink = genericShrink
 
 instance ( Arbitrary SscPayload
          , Arbitrary SscProof
          , Bi Raw
-         , HasConfiguration
+         , HasProtocolConstants
+         , HasProtocolMagic
```

Comment:
While we're touching these constraints, can we drop the monomorphic `Arbitrary {SscPayload,SscProof}` instances? These defer constraint solving errors to the use-site instead of the definition-site.

Reply:
Unless... oh god... we're relying on orphan instances that aren't in scope :sweat:

Reply:
We should work on removing orphan instances.